modules/geo-layers/src/tile-layer/tile-2d-header.js

```diff
@@ -2,6 +2,8 @@
 import {log} from '@deck.gl/core';
 
 export default class Tile2DHeader {
+  #metadata;
```

Comment:
Use `private _metadata`

We currently do not use pound private fields anywhere else in the code base. I'm certainly open to introducing it though we should have some sort of consensus on when and where it should be used.